lib/MySQL_Session.cpp

```diff
@@ -1484,6 +1504,7 @@ int MySQL_Session::handler_again___status_RESETTING_CONNECTION() {
 	int rc=myconn->async_change_user(myds->revents);
 	if (rc==0) {
 		__sync_fetch_and_add(&MyHGM->status.backend_change_user, 1);
+		MyHGM->status.p_counter_array[p_hg_counter::com_backend_change_user]->Increment();
```

Comment:
There are a lot of lines of code like this on.
Incrementing a value in `MyHGM->status.xxx` , followed by incrementing the prometheus counter.
This causes the updates of two counters a bit everywhere.
It seems redundant, not performance friendly (more operations) and difficult to maintain (we should never forget to update two counters, and not one).
I would recommend reverting all this double counters, and instead updating the prometheus counters when the prometheus exporter is called.
We are already doing this for MySQL status variables.

Reply:
Yes, this was the first impl I started doing, for later doing the callback approach in the other modules because as you said looks more efficient and avoids duplication. This is all replaced now!